### PR TITLE
Emphasise that review needs to be continuous

### LINKED ISSUE #91: Direct review to be continuous
We should clarify that review should be iterative and continuous during development, rather than just a final sign off process. It's simpler to address issues early on.

### PR DESCRIPTION
**Pull request by foster999**

### CODE REVIEW

**Comment by foster999** [comment]
Some of my changes here aren't very clear or concise, so I'm keen to hear any suggestions for improving them!

File: book/peer_review.md
Comment (RuSnook) [inline book/peer_review.md]:
```suggestion
This practice encourages developers to consider why they are writing code in a particular way and to vocalise this ("programming out loud"). Additionally, it gives reviewers a chance to suggest improvements and question the author's approach as the code is written. Working in this way can be more efficient than reviewing code separately - issues are identified sooner, so they are easier to fix. Despite the upfront cost of two individuals writing the code, the resulting code is often higher quality and contains fewer bugs. 
```

File: book/peer_review.md
Comment (RuSnook) [inline book/peer_review.md]:
```suggestion
There is a much larger burden on the reviewer and any issues that are identified may take a lot of time to fix across the whole project. A reviewer might highlight that certain quality assurance practices have not been used - for example, there has not been enough documentation or automated testing in the project. It would take a substantial amount of effort to add documentation and testing for the whole project. If this was identified earlier, the improved practices could be to apply as the remaining code is developed.
```

**Review by RuSnook — CHANGES_REQUESTED** [review]
grammar: 'are' missing

Looks good. Difficult to make more concise as there is a lot of detail to convey. I'll keep in mind as I work on issues and might come back to this page later.

**Comment by foster999** [comment]
I've added a bit more content here and restructured the page to use active headings as per #86. I haven't finished yet - I'd like to cut down the "Why" section and rephrase other parts to emphasise the action in the headings

File: book/peer_review.md
Comment (RuSnook) [inline book/peer_review.md:71]: Really like this

File: book/peer_review.md
Comment (RuSnook) [inline book/peer_review.md:152]: yes, I like this rewording.

File: book/peer_review.md
Comment (RuSnook) [inline book/peer_review.md]:
```suggestion
There is a much larger burden on the reviewer and any issues that are identified may take a lot of time to fix across the whole project. A reviewer might highlight that certain quality assurance practices have not been used - for example, there has not been enough documentation or automated testing in the project. It would take a substantial amount of effort to add documentation and testing for the whole project. If this was identified earlier, the improved practices could be applied as the remaining code is developed.
```

**Review by RuSnook — CHANGES_REQUESTED** [review]
sorry, my previous review suggestion didn't quite make sense.

**Comment by foster999** [comment]
Biggest changes were around what are now the first two sections. I'm still not totally happy with them, so keen to have a second opinion please!

File: book/peer_review.md
Comment (RuSnook) [inline book/peer_review.md]:
```suggestion
As we discussed in [](readable_code.md), good quality code is easier to read, understand and maintain. Peer review improves the quality of our code through the constructive challenges of the reviewer. As a reviewer, you might do this by suggesting alternative ways to represent the analysis or by asking about decisions that have been made in the approach to the analysis.
```

File: book/peer_review.md
Comment (RuSnook) [inline book/peer_review.md]:
```suggestion
Internal review should be carried out regularly within the development team. Reviewing code written by those with more and less experience than you is beneficial for both reviewer and developer. Similar questions can be asked from both perspectives, for the reviewer to get a good understanding of the approach and decisions behind the analysis.
```

**Review by RuSnook — CHANGES_REQUESTED** [review]

**Review by RuSnook — APPROVED** [review]
I've updated the 2 grammatically issues I found. I think it looks good. But will request review from Dan for another set of eyes.

File: book/peer_review.md
Comment (rowanhemsi) [inline book/peer_review.md]: Do we explain anywhere what the Aqua book is and that analysts are required to follow it? Also consider putting Aqua book quote next to this paragraph based on Aqua book

File: book/peer_review.md
Comment (rowanhemsi) [inline book/peer_review.md]: Not immediately obvious this is the quote source and not a header or relevant link

File: book/peer_review.md
Comment (rowanhemsi) [inline book/peer_review.md]: Consider rewording last sentence in paragraph for ease of reading

File: book/peer_review.md
Comment (rowanhemsi) [inline book/peer_review.md]: Consider replacing word "comprehensive"? There are things that that don't necessarily belong in the checklists that could still be considered good practice.
Also second sentence is a bit of a mouthful

File: book/peer_review.md
Comment (rowanhemsi) [inline book/peer_review.md]: All number bullet points in this section are showing as "1." when I build this locally

File: book/peer_review.md
Comment (rowanhemsi) [inline book/peer_review.md:48]: Do we talk about user needs elsewhere in the Duck Book? If not, might be worth making this a subsection in it's own right rather than an explanation of a bullet point on configs

File: book/peer_review.md
Comment (rowanhemsi) [inline book/peer_review.md:58]: Particularly like this comment

File: book/peer_review.md
Comment (rowanhemsi) [inline book/peer_review.md:64]: Like this too

File: book/peer_review.md
Comment (rowanhemsi) [inline book/peer_review.md:132]: Do we want to really drive it home and explicitly advise against waiting until the end of development for review?

Comment (rowanhemsi) [reply book/peer_review.md:132]: Ignore me, you've done this two sections down

File: book/peer_review.md
Comment (rowanhemsi) [inline book/peer_review.md]: FYI I'd never known where to find the link for this case study until looking at the raw markdown, but this might be a me problem 😅

**Review by rowanhemsi — COMMENTED** [review]
Some very picky feedback that you can choose to ignore, but generally I like this update

File: book/peer_review.md
Comment (foster999) [reply book/peer_review.md]: We do mention the responsibilities around Aqua in the Principles section, but it could probably be emphasised. I'm not sure if that's our role?

I've swapped the quotes, as it does make more sense to keep the Aqua one with the related paragraph 👍🏻

File: book/peer_review.md
Comment (foster999) [reply book/peer_review.md:48]: I agree that this is a much bigger problem - we need to highlight that it's a big part of the code of practice

File: book/peer_review.md
Comment (foster999) [reply book/peer_review.md]: Changed these to `epigraph` objectives, which look much clearer

File: book/peer_review.md
Comment (foster999) [reply book/peer_review.md]: Changed to subheadings

File: book/peer_review.md
Comment (rowanhemsi) [inline book/peer_review.md:7]: This formatting looks much clearer. Do we need to update quotes elsewhere in the book to match?
Also consider including book subtitle for context (The Aqua Book: guidance on producing quality analysis for government)

**Review by rowanhemsi — COMMENTED** [review]
This looks good, thanks for addressing my earlier feedback

Only other thing is a general reminder of the [Hemingway app](https://hemingwayapp.com/), which can help pick up readability issues like a couple of the wordier sentences I flagged before

File: book/peer_review.md
Comment (foster999) [reply book/peer_review.md:7]: Thanks, I've created new issues to address these more general points

**Pull request merged**
